api-gateway/src/main/java/org/swasth/apigateway/filters/HCXValidationFilter.java

```diff
@@ -58,10 +60,16 @@ public GatewayFilter apply(Config config) {
             String apiCallId = null;
             try {
                 StringBuilder cachedBody = new StringBuilder(StandardCharsets.UTF_8.decode(((DataBuffer) exchange.getAttribute(CACHED_REQUEST_BODY_ATTR)).asByteBuffer()));
-                Request request = new Request(JSONUtils.deserialize(cachedBody.toString(), HashMap.class));
-                correlationId = request.getCorrelationId();
-                apiCallId = request.getApiCallId();
-                request.validate(getMandatoryHeaders(), getDetails(request.getSenderCode()), getDetails(request.getRecipientCode()), getSubject(exchange), timestampRange);
+                Map<String,Object> requestBody = JSONUtils.deserialize(cachedBody.toString(), HashMap.class);
+                if(requestBody.containsKey(Constants.PAYLOAD)) {
+                    Request request = new Request(requestBody);
+                    correlationId = request.getCorrelationId();
+                    apiCallId = request.getApiCallId();
+                    request.validate(getMandatoryHeaders(), getDetails(request.getSenderCode()), getDetails(request.getRecipientCode()), getSubject(exchange), timestampRange);
+                } else {
+                    ErrorRequest request = new ErrorRequest(requestBody);
+                    request.validate(getErrorMandatoryHeaders(), getDetails((String) requestBody.get(Constants.SENDER_CODE)),getDetails((String) requestBody.get(Constants.RECIPIENT_CODE)), getSubject(exchange));
+                }
```

Comment:
Rename the "Request" class as "JWERequest". 
Rename the "ErrorRequest" class as "PlainRequest".

PlainRequst class should have implementation of getSenderCode, getRecipientCode methods.

Reply:
Done